Cargo.toml

```diff
@@ -31,7 +31,8 @@ url = "2"
 atomic-option = "0.1"
 fnv = "1.0.5"
 native-tls = { version = "0.2", optional = true }
-clap = { version = "3.1.0", optional = true }
+openssl = { version = "0.10.61", optional = true }
```

Comment:
Why do we now need a direct dependency on `openssl`?

Reply:
direct dependency with version pinned will prevent the reset of openssl to 0.9.xx in 'minimal versions job' which will error.  From the latest workflow [run](https://github.com/jonhoo/faktory-rs/actions/runs/7237492281/job/19717219623):
![image](https://github.com/jonhoo/faktory-rs/assets/117771945/c8efa10d-5cf5-48af-a9d4-4381dabe0401)

If we are ok with that job failing, this direct dependency is redundant. 

There is also probably a better way out, which I am missing..

Reply:
This should now be fixed by https://github.com/jonhoo/faktory-rs/pull/40